### ISSUE TITLE
Delete permission, does not clear caches.

### ISSUE THREAD

**Issue by jcoyne** [projecthydra/sufia#3033]
When you delete access from a work, it doesn't detect the permissions change.
```ruby
"permissions_attributes"=><ActionController::Parameters {"1"=><ActionController::Parameters 
{"access"=>"read", "id"=>"715ac526-3d88-4b4b-838a-7c748a42bb14/5e/b5/93
/a8/5eb593a8-9874-48dd-9847-681fb69372ea", "_destroy"=>"true"} permitted: true>
```

Where the id points at a `Hydra::AccessControls::Permission`
```ruby
obj = ActiveFedora::Base.find('715ac526-3d88-4b4b-838a-7c748a42bb14/5e/b5/93
/a8/5eb593a8-9874-48dd-9847-681fb69372ea')
<Hydra::AccessControls::Permission id: 715ac526-3d88-4b4b-838a-7c748a42bb14/5e/b5/93/a8/5eb593a8-9874-48dd-9847-681fb69372ea agent: "http://projecthydra.org/ns/auth/person#cam156@psu.edu" mode: "http://www.w3.org/ns/auth/acl#Read" access_to: "6m311q35h">
```

After this is deleted, the `curation_concern.permissions` still returns it:
```ruby
curation_concern.permissions
#<Hydra::AccessControl::CollectionRelationship:0x007f9064d768a0 @owner=#
<Hydra::AccessControl id: "715ac526-3d88-4b4b-838a-7c748a42bb14">, @relationship=
[<Hydra::AccessControls::Permission id: 715ac526-3d88-4b4b-838a-7c748a42bb14/aa/d9/bf
/29/aad9bf29-422f-4870-9299-53b045cbbb2d agent: "http://projecthydra.org/ns/auth
/person#jcoyne85@stanford.edu" mode: "http://www.w3.org/ns/auth/acl#Write" access_to: "6m311q35h">,
<Hydra::AccessControls::Permission id: 715ac526-3d88-4b4b-838a-7c748a42bb14/5e/b5/93/a8/5eb593a8-9874-48dd-9847-681fb69372ea agent: "http://projecthydra.org/ns/auth/person#cam156@psu.edu" mode: "http://www.w3.org/ns/auth/acl#Read" access_to: "6m311q35h">,
<Hydra::AccessControls::Permission id: 715ac526-3d88-4b4b-838a-7c748a42bb14/16/7f/8c/0e/167f8c0e-8393-4701-86d8-17705a5ab7d6 agent: "http://projecthydra.org/ns/auth/group#registered" mode: "http://www.w3.org/ns/auth/acl#Read" access_to: "6m311q35h">]>
```

Possibly `AccessControl#relationship` (aka. `#permissions`) needs clearing. This seems to be an AF::Collection relationship, so perhaps calling `reset` will work.

**Comment by jcoyne** [comment]
I suspect this is the cause of #2820

**Comment by jcoyne** [comment]
Closed by #3036
